pkg/encryption/chunk_encryption.go

```diff
@@ -0,0 +1,37 @@
+package encryption
+
+import (
+	"github.com/ethersphere/bee/pkg/swarm"
+	"golang.org/x/crypto/sha3"
+)
+
+// ChunkEncrypter encrypts chunk data.
+type ChunkEncrypter interface {
+	EncryptChunk([]byte) (key Key, encryptedSpan, encryptedData []byte, err error)
+}
+
+type ce struct{}
+
+func NewChunkEncrypter() ChunkEncrypter { return &ce{} }
+
+func (c *ce) EncryptChunk(chunkData []byte) (Key, []byte, []byte, error) {
+	key := GenerateRandomKey(KeyLength)
+	encryptedSpan, err := newSpanEncryption(key).Encrypt(chunkData[:8])
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	encryptedData, err := newDataEncryption(key).Encrypt(chunkData[8:])
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	return key, encryptedSpan, encryptedData, nil
+}
+
+func newSpanEncryption(key Key) Interface {
+	refSize := int64(swarm.HashSize + KeyLength)
+	return New(key, 0, uint32(swarm.ChunkSize/refSize), sha3.NewLegacyKeccak256)
```

Comment:
We seem to be increasing number of uses of `sha3.NewLegacyKeccak256` (2 here).

It might be prudent at some point to switch to specific other one. Since other `swarm` constant is used, maybe we could use `swarm.NewHasher` intead.

Reply:
This was already thoroughly discussed. As you may notice, it is the official golang `sha3` library that has this name. This is the keccak256 standard that is defined as the sha3 flavor we're using and that's how they call it. I'm not sure if we should wrap or try to alter this. It came up several times but I'm not sure if and what we should do with this.